File: frontend3/src/api/trial.ts
import apiClient from './client'
import { API_ENDPOINTS } from '@/config/constants'

// Trial Types
export interface StartTrialRequest {
  game_id?: string
}

export interface TrialProfile {
  id: string
  username: string
  balance: number
  total_spins: number
  total_wagered: number
  total_won: number
  is_trial: boolean
}

export interface TrialResponse {
  session_token: string
  expires_at: number
  player: TrialProfile
}

export interface TrialBalanceResponse {
  balance: number
}

export const trialApi = {
  // Start a new trial session (no auth required)
  async startTrial(gameId?: string): Promise<TrialResponse> {
    const response = await apiClient.post<TrialResponse>(
      API_ENDPOINTS.AUTH.TRIAL,
      gameId ? { game_id: gameId } : {}
    )
    return response.data
  },

  // Get trial profile (requires trial token)
  async getProfile(): Promise<TrialProfile> {
    const response = await apiClient.get<TrialProfile>(API_ENDPOINTS.TRIAL.PROFILE)
    return response.data
  },

  // Get trial balance (requires trial token)
  async getBalance(): Promise<TrialBalanceResponse> {
    const response = await apiClient.get<TrialBalanceResponse>(API_ENDPOINTS.TRIAL.BALANCE)
    return response.data
  },
}

File: backend/internal/api/dto/trial.go
package dto

// StartTrialRequest represents the request to start a trial session
type StartTrialRequest struct {
	GameID string `json:"game_id,omitempty"` // Optional game ID
}

// TrialProfile represents trial player profile info
type TrialProfile struct {
	ID           string  `json:"id"`
	Username     string  `json:"username"`
	Balance      float64 `json:"balance"`
	TotalSpins   int     `json:"total_spins"`
	TotalWagered float64 `json:"total_wagered"`
	TotalWon     float64 `json:"total_won"`
	IsTrial      bool    `json:"is_trial"`
}

// TrialResponse represents the response when starting a trial session
type TrialResponse struct {
	SessionToken string       `json:"session_token"`
	ExpiresAt    int64        `json:"expires_at"` // Unix timestamp
	Player       TrialProfile `json:"player"`
}

// TrialBalanceResponse represents the response for balance query
type TrialBalanceResponse struct {
	Balance float64 `json:"balance"`
}
